### PR TITLE
PIMOB- 2035: Support SPM

### PR DESCRIPTION
**Pull request by ghost**
- 67k files is from JOSESWift.xcframework
- This is workaround using wrapper to have binary targets 
- dummy.swift is apart for this workaround

### CODE REVIEW

**Review by okhan-okbay-cko — DISMISSED** [review]

**Review by okhan-okbay-cko — APPROVED** [review]

**Comment by okhan-okbay-cko** [comment]
this might do well with a squash commit before merging

File: Checkout3DSPackages/dummy.swift [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (deepesh-vasthimal-cko) [inline Checkout3DSPackages/dummy.swift:1]: Is there any online reference to why we are adding this dummy swift file and its purpose ?

File: Dependencies/JOSESwift.xcframework/Info.plist [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (deepesh-vasthimal-cko) [inline Dependencies/JOSESwift.xcframework/Info.plist:1]: Can you automate the creation of xcframework of JoseSwift for future upgrades if required?

File: Package.swift [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    products: [
+        .library(
+            name: "Checkout3DSPackages",
+            targets: ["Checkout3DSPackages"]
+        )
+    ],
+    dependencies: [
+        .package(
+            url: "https://github.com/checkout/checkout-event-logger-ios-framework.git",
+            from: "1.2.4"
```
Comment (deepesh-vasthimal-cko) [inline Package.swift:18]: Test : What happens with different version of checkout-event-logger-ios-framework with Checkout3DSPackages and Frames

Comment (deepesh-vasthimal-cko) [reply Package.swift:18]: So what happens if Frames is not compatible with 1.2.4?. If Checkout3DSPackages is configured to use the latest Does it force Frames use the latest version?

File: Dependencies/JOSESwift.xcframework/Info.plist (hunk 2 above)
Comment (deepesh-vasthimal-cko) [reply Dependencies/JOSESwift.xcframework/Info.plist:1]: Thanks, In the meantime can you please document the steps if we need to upgrade the JoseSwift manually till we have automation in place?. Not a blocker for PR.

**Review by deepesh-vasthimal-cko — APPROVED** [review]

File: .gitignore [hunk 4]
```diff
@@ -1,2 +1,158 @@
 
+# Created by https://www.toptal.com/developers/gitignore/api/macos,swift,xcode,carthage,cocoapods,swiftpm
+# Edit at https://www.toptal.com/developers/gitignore?templates=macos,swift,xcode,carthage,cocoapods,swiftpm
+
```
Comment (okhan-okbay-cko) [inline .gitignore:4]: this has been forgotten

File: .gitignore [hunk 5]
```diff
@@ -1,2 +1,158 @@
...
+.VolumeIcon.icns
+.com.apple.timemachine.donotpresent
+
+# Directories potentially created on remote AFP share
+.AppleDB
+.AppleDesktop
+Network Trash Folder
+Temporary Items
+.apdisk
+
```
Comment (okhan-okbay-cko) [inline .gitignore:49]: I think these all are irrelevant to mobile development? Can we use the similar ones that we have for Frames or any other repos?

**Review by deepesh-vasthimal-cko — CHANGES_REQUESTED** [review]
Please check with JoseSwift about XCframework creation and re-distribution through our SDKs.

File: .gitignore [hunk 6]
```diff
@@ -1,2 +1,158 @@
...
+### Xcode Patch ###
+*.xcodeproj/project.workspace/xcshareddata/swiftpm/Package.resolved
+*.xcodeproj/*
+!*.xcodeproj/project.pbxproj
+*/IDEWorkspaceChecks.plist
+!*.xcworkspace/contents.xcworkspacedata
+**/xcshareddata/WorkspaceSettings.xcsettings
+
+# End of https://www.toptal.com/developers/gitignore/api/macos,swift,xcode,carthage,cocoapods,swiftpm
```
Comment (okhan-okbay-cko) [inline .gitignore:158]: also we can add an empty line at the end since it seems lacking

Comment (okhan-okbay-cko) [inline .gitignore:158]: this line might be removed directly

File: .gitignore [hunk 7]
```diff
@@ -1,2 +1,158 @@
...
+# https://docs.fastlane.tools/best-practices/source-control/#source-control
+
+fastlane/report.xml
+fastlane/Preview.html
+fastlane/screenshots/**/*.png
+fastlane/test_output
+
+# Code Injection
+# After new code Injection tools there's a generated folder /iOSInjectionProject
+# https://github.com/johnno1962/injectionforxcode
```
Comment (okhan-okbay-cko) [inline .gitignore:128]: these look irrelevant

File: .gitignore (hunk 4 above)
Comment (okhan-okbay-cko) [reply .gitignore:4]: I think there are differences. Also we better not have any other website references if we don't have to

**Review by okhan-okbay-cko — APPROVED** [review]

**Review by okhan-okbay-cko — APPROVED** [review]
Looks good to me 👍

File: README.md [hunk 8]
```diff
@@ -262,7 +262,7 @@ After initiating the authentication process and obtaining the `AuthenticationRes
 
 Our iOS SDK depends on some external libraries:
 
-* To help maintain security, we use [JOSESwift](https://github.com/airsidemobile/JOSESwift).
+* To help maintain security, we use [JOSESwift](https://github.com/airsidemobile/JOSESwift). To overcome the limitation in Swift Package Manager (SPM) related to binary target dependencies, we've integrated JOSESwift as an xcframework sourced from the same [JOSESwift](https://github.com/airsidemobile/JOSESwift).
```
Comment (deepesh-vasthimal-cko) [inline README.md:265]: 👍

**Review by deepesh-vasthimal-cko — DISMISSED** [review]

File: Dependencies/JOSESwift.xcframework/LICENSE [hunk 9]
```diff
@@ -0,0 +1,176 @@
+Apache License
+Version 2.0, January 2004
+http://www.apache.org/licenses/
+
+TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTION
+
```
Comment (okhan-okbay-cko) [inline Dependencies/JOSESwift.xcframework/LICENSE:6]: would it be good to add this as well? 
Screenshot 2023-10-20 at 14 50 10

Comment (deepesh-vasthimal-cko) [reply Dependencies/JOSESwift.xcframework/LICENSE:6]: Does Apache License tell us to do that?

Comment (okhan-okbay-cko) [reply Dependencies/JOSESwift.xcframework/LICENSE:6]: As far as I interpret it, it might be good to have this under the `Dependencies` folder. This should be the `NOTICE` that's mentioned in the license file
